ui/src/components/DataTable/index.jsx

```diff
@@ -40,22 +41,39 @@ export default class DataTable extends Component {
     renderRow: func.isRequired,
     /**
      * A function to execute when a column header is clicked.
-     * Will receive a single argument which is the column name.
+     * Will receive a single argument
+     * which is the header object for the corresponding column.
      * This can be used to handle sorting.
      */
     onHeaderClick: func,
     /**
      * A header name to sort on.
      */
-    sortByHeader: string,
+    sortByLabel: string,
     /**
      * The sorting direction.
      */
     sortDirection: oneOf(['desc', 'asc']),
     /**
-     * A list of header names to use on the table starting from the left.
+     * A list of header objects to use on the table starting from the left.
      */
-    headers: arrayOf(string),
+    headers: arrayOf(
+      shape({
+        /**
+         * A string identifying the data type of the column contents
+         * (e.g., 'string', 'object', 'number'). */
+        id: string,
```

Comment:
The comment for `id ` seems to belong to `type`.

Reply:
Sorry about the mixup, I'll get that sorted.